File: airline-integration-service/internal/api/models/api_response.go
package models

type ApiResponse struct {
	StatusCode    *int    `json:"status_code,omitempty"`
	StatusMessage string  `json:"status_message,omitempty"`
	Description   *string `json:"description,omitempty"`
	Data          *any    `json:"data,omitempty"`
	Took          *int64  `json:"took,omitempty"`
}

// func NewApiResponse(statusCode int, statusMessage string) *ApiResponse {
// 	return &ApiResponse{
// 		StatusCode: &statusCode,
// 		StatusMessage: statusMessage,
// 	}
// }

func NewErrorHandler(statusCode int, statusMessage string, description string) *ApiResponse {
	return &ApiResponse{
		StatusCode:    &statusCode,
		StatusMessage: statusMessage,
		Description:   &description,
	}
}

func NewApiResponse(statusCode int, statusMessage string, description string, data any, took int64) *ApiResponse {
	return &ApiResponse{
		StatusCode:    &statusCode,
		StatusMessage: statusMessage,
		Description:   &description,
		Data:          &data,
		Took:          &took,
	}
}

File: aerona-base/src/main/java/com/thanhtd/aerona/base/core/APIResponse.java
package com.thanhtd.aerona.base.core;

import com.thanhtd.aerona.base.constant.ErrorCode;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class APIResponse {
    private Integer statusCode;

    private String statusMessage;

    private String description;

    private Object data;

    private Long took;

    public APIResponse(ErrorCode errorCode, String description, Long took, Object data) {
        this.statusCode = errorCode.getValue();
        this.statusMessage = errorCode.getMessage();
        this.description = description;
        this.data = data;
        this.took = took;
    }
}
